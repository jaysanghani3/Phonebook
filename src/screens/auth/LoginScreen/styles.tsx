import {StyleSheet} from 'react-native';
import Colors from '../../../utils/Colors';
const styles = StyleSheet.create({
  welcomeHeader: {
    fontSize: 30,
    fontWeight: 'bold',
    color: Colors.appColor,
  },
  forgetText: {
    color: Colors.appColor,
    textAlign: 'right',
  },
  formBox: {
    marginVertical: 40,
  },
  fieldArea: {
    marginVertical: 10,
  },
  textArea: {
    paddingStart: 10,
    // display: 'block',
    borderRadius: 8,
    borderColor: Colors.gray,
    borderWidth: 1,
    backgroundColor: Colors.white,
    flexDirection: 'row',
  },
  eyeContainer: {
    padding: 20,
  },
  eye: {
    width: 25,
    height: 25,
    position: 'absolute',
    right: 15,
    top: 11,
  },
});

export default styles;
