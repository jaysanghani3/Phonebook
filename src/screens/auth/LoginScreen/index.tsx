import React, {useState} from 'react';
import {
  Text,
  View,
  TextInput,
  TouchableOpacity,
  Image,
  Keyboard,
} from 'react-native';
import {StackActions} from '@react-navigation/native';
import styles from './styles';
import globalStyles from '../../../utils/globalStyles';
import Loader from '../../../components/Loader';
import Icons from '../../../assets';
import Strings from '../../../translations/Strings';
import Colors from '../../../utils/Colors';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import {RouteStackParamsList} from '../../../navigators/AppNavigators';
import {Routes} from '../../../navigators/routes';
import auth from '@react-native-firebase/auth';
import Toast from 'react-native-simple-toast';

interface userCredType {
  email: string;
  password: string;
}

interface errType {
  email?: string;
  password?: string;
}

type LoginScreenProps = NativeStackScreenProps<
  RouteStackParamsList,
  Routes.LoginScreen
>;

const LoginScreen = ({navigation}: LoginScreenProps) => {
  const [userCreditentials, setUserCreditentials] = useState<userCredType>({
    email: '',
    password: '',
  });
  const [isPasswordVisible, setIsPasswordVisible] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState<errType>({email: '', password: ''});

  const handleLogin = async () => {
    Keyboard.dismiss();
    try {
      if (!userCreditentials.email && !userCreditentials.password) {
        setError({
          email: Strings.required,
          password: Strings.required,
        });
        return;
      }
      if (!userCreditentials.email) {
        setError({email: Strings.required});
        return;
      }
      if (!userCreditentials.password) {
        setError({password: Strings.required});
        return;
      }
      if (
        !userCreditentials.email.includes('@') ||
        !userCreditentials.email.includes('.')
      ) {
        setError({email: Strings.invalidEmail});
        return;
      }
      setError({email: '', password: ''});
      setIsLoading(true);
      const isUserLoggedIn = await auth().signInWithEmailAndPassword(
        userCreditentials.email,
        userCreditentials.password,
      );
      setIsLoading(false);
      if (isUserLoggedIn.user.emailVerified) {
        navigation.dispatch(StackActions.replace('HomeScreen'));
      } else {
        await auth().currentUser?.sendEmailVerification();
        await auth().signOut();
        Toast.show(Strings.verifyEmail, Toast.LONG);
      }
    } catch (e) {
      const err: string = (e as any).code.split('/')[1].split('-').join(' ');
      Toast.show(err, Toast.LONG);
      setIsLoading(false);
    }
  };

  const handleForgetPassword = () =>
    navigation.navigate(Routes.ForgetPasswordScreen);

  const handelSignup = () => navigation.navigate(Routes.SignupScreen);
  return (
    <View style={globalStyles.container}>
      <Loader state={isLoading} />
      <View>
        <Text style={styles.welcomeHeader}>{Strings.welcomeText}</Text>
        <Text>{Strings.enterYourDetails}</Text>
      </View>
      <View style={styles.formBox}>
        <View style={styles.fieldArea}>
          <View style={styles.textArea}>
            <TextInput
              defaultValue={userCreditentials.email}
              onChangeText={email =>
                setUserCreditentials({...userCreditentials, email})
              }
              inputMode="email"
              placeholder={Strings.emailPlaceholder}
              placeholderTextColor={Colors.gray}
              style={globalStyles.flex1}
            />
          </View>
          <Text style={globalStyles.error}>{error.email}</Text>
        </View>
        <View style={styles.fieldArea}>
          <View style={styles.textArea}>
            <TextInput
              style={globalStyles.flex1}
              defaultValue={userCreditentials.password}
              secureTextEntry={!isPasswordVisible}
              onChangeText={password =>
                setUserCreditentials({...userCreditentials, password})
              }
              placeholder={Strings.passwordPlaceholder}
            />
            <TouchableOpacity
              style={styles.eyeContainer}
              onPress={() => setIsPasswordVisible(!isPasswordVisible)}>
              <Image
                source={!isPasswordVisible ? Icons.show : Icons.hide}
                style={styles.eye}
              />
            </TouchableOpacity>
          </View>
          <Text style={globalStyles.error}>{error.password}</Text>
        </View>
        <Text style={styles.forgetText} onPress={handleForgetPassword}>
          {Strings.forgotText}
        </Text>
      </View>
      <View>
        <TouchableOpacity onPress={handleLogin} style={globalStyles.btn}>
          <Text style={globalStyles.btnText}>{Strings.loginBtn}</Text>
        </TouchableOpacity>
      </View>
      <View style={globalStyles.topBorder}>
        <Text style={globalStyles.textCenter}>
          {Strings.dontHaveAccount}
          <Text style={globalStyles.appColorText} onPress={handelSignup}>
            {Strings.signupBtn}
          </Text>
        </Text>
      </View>
    </View>
  );
};

export default LoginScreen;
