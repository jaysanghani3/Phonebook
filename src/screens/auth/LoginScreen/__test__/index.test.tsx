import renderer from 'react-test-renderer';
import LoginScreen from '../index';

it('renders correctly', () => {
  const tree = renderer
    .create(<LoginScreen navigation={undefined} route={undefined} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
