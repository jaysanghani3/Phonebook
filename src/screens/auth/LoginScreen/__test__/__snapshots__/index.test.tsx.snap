// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "center",
      "padding": 25,
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#66347f",
          "fontSize": 30,
          "fontWeight": "bold",
        }
      }
    >
      Welcome
    </Text>
    <Text>
      Enter your details to log in.
    </Text>
  </View>
  <View
    style={
      {
        "marginVertical": 40,
      }
    }
  >
    <View
      style={
        {
          "marginVertical": 10,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#8D8D8D",
            "borderRadius": 8,
            "borderWidth": 1,
            "flexDirection": "row",
            "paddingStart": 10,
          }
        }
      >
        <TextInput
          defaultValue=""
          inputMode="email"
          onChangeText={[Function]}
          placeholder="Email"
          placeholderTextColor="#8D8D8D"
          style={
            {
              "flex": 1,
            }
          }
        />
      </View>
      <Text
        style={
          {
            "color": "#FF0000",
            "fontSize": 12,
            "marginStart": 10,
          }
        }
      />
    </View>
    <View
      style={
        {
          "marginVertical": 10,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#8D8D8D",
            "borderRadius": 8,
            "borderWidth": 1,
            "flexDirection": "row",
            "paddingStart": 10,
          }
        }
      >
        <TextInput
          defaultValue=""
          onChangeText={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          style={
            {
              "flex": 1,
            }
          }
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "padding": 20,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../src/assets/show.png",
              }
            }
            style={
              {
                "height": 25,
                "position": "absolute",
                "right": 15,
                "top": 11,
                "width": 25,
              }
            }
          />
        </View>
      </View>
      <Text
        style={
          {
            "color": "#FF0000",
            "fontSize": 12,
            "marginStart": 10,
          }
        }
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        {
          "color": "#66347f",
          "textAlign": "right",
        }
      }
    >
      Forgot Password
    </Text>
  </View>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#66347f",
          "borderRadius": 8,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "borderTopColor": "#8D8D8D",
        "borderTopWidth": 1,
        "marginTop": 50,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "textAlign": "center",
        }
      }
    >
      Don't have an account? 
      <Text
        onPress={[Function]}
        style={
          {
            "color": "#66347f",
          }
        }
      >
        Sign Up
      </Text>
    </Text>
  </View>
</View>
`;
